lib/Services/GrantService.php

```diff
@@ -28,27 +28,49 @@ class GrantService implements GrantRoutes
 
     private PendingGrantTransformer $pendingTransformer;
 
+    private string $clientUrl = '';
+
     /**
      * GrantService constructor.
      *
      * @param  GrantValidator|null  $validator
      */
     public function __construct(
         ApiClient $apiClient,
+        string $clientUrl = '',
         ?GrantTransformer $grantTransformer = null,
         ?PendingGrantTransformer $pendingTransformer = null
     ) {
         $this->apiClient = $apiClient;
+        $this->clientUrl = $clientUrl;
         $this->validator = new GrantValidator;
         $this->grantTransformer = $grantTransformer ?? new GrantTransformer;
-        $this->pendingTransformer = $pendingTransformer ?? new PendingGrantTransformer;
+        $this->pendingTransformer
+            = $pendingTransformer ?? new PendingGrantTransformer;
+    }
+
+    public function addClientToStruct(array $grantStructure): array
```

Comment:
Instead of copying the whole structure again, what do you think about accepting a reference to the grant object?

```suggestion
    public function injectClientUrl(array &$grantStructure): array
```